lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -2028,19 +2028,16 @@ def build_networks(self, docker_networks):
     def build_endpoint_spec(self):
         endpoint_spec_args = {}
         if self.publish is not None:
-            ports = {}
+            ports = []
             for port in self.publish:
+                port_spec = {
+                  'Protocol' : port['protocol'],
```

Comment:
```suggestion
                    'Protocol': port['protocol'],
```
You need to indent by four spaces, and remove the space before `:`. (Same for the next two lines.)